src/overlays/actors/ovl_Bg_Breakwall/z_bg_breakwall.c

```diff
@@ -53,21 +66,268 @@ static ColliderQuadInit D_80870760 = {
+                    actor->velocity.y = Rand_ZeroOne() + (accel * 0.6f);
+                    actor->world.rot.y += (s16)((Rand_ZeroOne() - 0.5f) * 3000.0f);
+                    actor->world.rot.x = (s16)(Rand_ZeroOne() * 3500.0f) + 2000;
+                    actor->world.rot.z = (s16)(Rand_ZeroOne() * 3500.0f) + 2000;
+                    actor->parent = &this->dyna.actor;
+                    actor->scale.x = actorScaleList[k].x + Rand_CenteredFloat(0.001f);
+                    actor->scale.y = actorScaleList[k].y + Rand_CenteredFloat(0.001f);
+                    actor->scale.z = actorScaleList[k].z + Rand_CenteredFloat(0.001f);
+                }
+            }
+        }
+    }
+
+    return actor;
+}
+
+/**
+ * Sets up the collision model as well is the object dependency and action function to use.
+ */
+void BgBreakwall_SetupWait(BgBreakwall* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->bankIndex)) {
+        CollisionHeader* colHeader = NULL;
+        s32 wallType = ((this->dyna.actor.params >> 13) & 3) & 0xFF;
+
+        this->dyna.actor.objBankIndex = this->bankIndex;
+        Actor_SetObjectDependency(globalCtx, &this->dyna.actor);
+        this->dyna.actor.flags &= ~0x10;
+        this->dyna.actor.draw = BgBreakwall_Draw;
+        CollisionHeader_GetVirtual(sBombableWallInfo[wallType].colHeader, &colHeader);
+        this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+
+        if (wallType == BWALL_KD_LAVA_COVER) {
+            BgBreakwall_SetupAction(this, BgBreakwall_LavaCoverMove);
+        } else {
+            BgBreakwall_SetupAction(this, BgBreakwall_Wait);
+        }
+    }
+}
+
+/**
+ * Checks for an explosion using quad collision. If the wall or floor is exploded then it will spawn fragments and
+ * despawn itself.
+ */
+void BgBreakwall_Wait(BgBreakwall* this, GlobalContext* globalCtx) {
+    if (this->collider.base.acFlags & 2) {
+        Vec3f effectPos;
+        s32 wallType = ((this->dyna.actor.params >> 13) & 3) & 0xFF;
+
+        DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+
+        effectPos.x = 0.0f;
+        effectPos.z = 0.0f;
+        effectPos.y = 0.0f;
+
+        if (this->dyna.actor.world.rot.x == 0) {
+            effectPos.y = 55.0f;
+        } else {
+            effectPos.z = 25.0f;
+            effectPos.y = -10.0f;
+        }
+
+        BgBreakwall_SpawnFragments(globalCtx, this, &effectPos, 0.0f, 6.4f, 5.0f, 1, 2.0f);
+        Flags_SetSwitch(globalCtx, this->dyna.actor.params & 0x3F);
+
+        if (wallType == BWALL_KD_FLOOR) {
+            Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_EXPLOSION);
+        } else {
+            Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_WALL_BROKEN);
+        }
+
+        if (wallType == BWALL_DC_ENTRANCE) {
+            if (!(Flags_GetEventChkInf(0xB0))) {
```

Comment:
```suggestion
        if ((wallType == BWALL_DC_ENTRANCE) && !Flags_GetEventChkInf(0xB0)) {
```
And fix the braces/indentation